### ISSUE TITLE
Training loss suddenly increase after a restart (adam optimizer)

### ISSUE THREAD

**Issue by Lingygao** [facebookresearch/fairseq#4066]
## ❓ Questions and Help

Hi there,

I am pre-training a RoBERTa language model with the command
```
fairseq-train --fp16 "train/LibriSpeech/" \
    --task masked_lm --criterion masked_lm \
    --save-dir "train/" \
    --keep-last-epochs 1 \
    --train-subset train \
    --arch roberta_base \
    --num-workers 2 \
    --optimizer adam --adam-betas '(0.9, 0.98)' --adam-eps 1e-06 --clip-norm 0.0 \
    --lr-scheduler polynomial_decay --lr 0.0005 --total-num-update 100000 --warmup-updates 10000 \
    --dropout 0.1 --attention-dropout 0.1 --weight-decay 0.01 \
    --sample-break-mode eos --tokens-per-sample 512 --max-positions 512 \
    --mask-prob 0.2  \
    --batch-size 16 --update-freq 64 --max-update 250000 \
    --seed 5 --log-format simple --log-interval 10 --skip-invalid-size-inputs-valid-test \
    --save-interval-updates 3600 --keep-interval-updates 4 \
    --validate-interval 10
  
```

After 50k updates, the training loss is around 2.3. I stopped and reinstalled the fairseq by building it from the source. Then I restarted the training process but I saw the training loss from the loaded last checkpoint was 3.0. 

I tested the restart checkpoint after 5k restart training updates on my downstream tasks. The results showed that the accuracy still increased as expected, compared to the last checkpoint before restarting.

## my question
I wonder if I made some mistakes or what is going on with the loss?

#### What's your environment?

 - fairseq Version (1.0) :
 - PyTorch Version (1.10.0)
 - OS (CentOS):
 - How you installed fairseq (git clone then pip install ./editable):
 - Python version: 3.7
 - CUDA/cuDNN version: 10.2
 - GPU models and configuration: V100
